src/darwin/Framework/CHIP/MTRDeviceController.mm

```diff
@@ -352,6 +352,7 @@ - (BOOL)isRunning
 
 - (BOOL)isSuspended
 {
+    std::lock_guard lock(_suspensionLock);
```

Comment:
Adding comment to avoid auto merge while I investigate other related flaky tests, or re-running successful runs.